crates/solver/src/settlement_rater.rs

```diff
@@ -259,3 +277,113 @@ impl SettlementRating for SettlementRater {
+        let payoff_obj_minus_cap = self.payoff(
+            objective.clone() - self.score_cap.clone(),
+            objective.clone(),
+            probability_success.clone(),
+            cost_fail.clone(),
+        );
+        tracing::trace!(?payoff_obj_minus_cap, "Payoff obj minus cap");
+        let payoff_cap = self.payoff(
+            self.score_cap.clone(),
+            objective.clone(),
+            probability_success.clone(),
+            cost_fail.clone(),
+        );
+        tracing::trace!(?payoff_cap, "Payoff cap");
+        let optimal_bid = if payoff_obj_minus_cap >= BigRational::zero()
+            && payoff_cap <= BigRational::zero()
+        {
+            Ok(probability_success * objective - probability_fail * cost_fail)
+        } else if payoff_obj_minus_cap >= BigRational::zero() && payoff_cap > BigRational::zero() {
+            Ok(objective
+                - probability_fail / probability_success * (self.score_cap.clone() + cost_fail))
+        } else if payoff_obj_minus_cap >= BigRational::zero() && payoff_cap > BigRational::zero() {
+            Ok(probability_success / probability_fail * self.score_cap.clone() - cost_fail)
+        } else {
+            Err(anyhow!("Invalid bid"))
+        };
+        tracing::trace!(?optimal_bid, "Optimal bid");
+        optimal_bid
+    }
+
+    fn payoff(
+        &self,
+        score_reference: BigRational,
+        objective: BigRational,
+        probability_success: BigRational,
+        cost_fail: BigRational,
+    ) -> BigRational {
+        tracing::trace!(
+            ?score_reference,
+            ?objective,
+            ?probability_success,
+            ?cost_fail,
+            "Computing payoff"
+        );
+        let probability_fail = BigRational::one() - probability_success.clone();
+        let payoff_success = min(objective - score_reference.clone(), self.score_cap.clone());
+        tracing::trace!(?payoff_success, "Payoff success");
+        let payoff_fail = -min(score_reference, self.score_cap.clone()) - cost_fail;
+        tracing::trace!(?payoff_fail, "Payoff fail");
+        let payoff_expectation =
+            probability_success * payoff_success + probability_fail * payoff_fail;
+        tracing::trace!(?payoff_expectation, "Payoff expectation");
+        payoff_expectation
+    }
+}
+
+mod tests {
+    #[test]
+    fn compute_score_with_success_probability_test() {
+        let objective_value = num::BigRational::from_float(251547381429604400.).unwrap();
+        let success_probability = 0.9202405649482063;
+        let score_cap = num::BigRational::from_float(10_000_000_000_000_000.).unwrap();
+        let score_calculator = super::ScoreCalculator::new(score_cap);
+        let score = score_calculator
+            .compute_score(&objective_value, success_probability)
+            .unwrap();
+        assert_eq!(score.score(), 250680657682686317u128.into());
+    }
```

Comment:
Three more examples would use
1) `objective_value = 1e16`, `success_probability = 0.9` which gives a score of `9e15`
2) `objective_value = 7e16`, `success_probability = 0.66666666666666666666` (i.e. 2/3) which gives a score of `9.5e16`
3) `objective_value = 1e16`, `success_probability = 0.33333333333333333333` (i.e. 1/3) which gives a score of `5e15`
(up to rounding errors)